pkg/dockerregistry/server/errorblobstore.go

```diff
@@ -127,3 +130,42 @@ func (f guardCreateOptions) Apply(v interface{}) error {
 	}
 	return nil
 }
+
+// statCrossMountCreateOptions ensures the expected options type is passed, and optionally pre-fills the cross-mount stat info
+type statCrossMountCreateOptions struct {
+	ctx            context.Context
+	sourceRepoName reference.Named
+	blobDigest     digest.Digest
+}
+
+var _ distribution.BlobCreateOption = statCrossMountCreateOptions{}
+
+func (f statCrossMountCreateOptions) Apply(v interface{}) error {
+	opts, ok := v.(*distribution.CreateOptions)
+	if !ok {
+		return fmt.Errorf("Unexpected create options: %#v", v)
+	}
+
+	if !opts.Mount.ShouldMount {
+		return nil
+	}
+
+	desc, err := statSourceRepository(f.ctx, f.sourceRepoName, f.blobDigest)
+	if err != nil {
+		context.GetLogger(f.ctx).Infof("cannot mount blob %s from repository %s: %v - disabling cross-repo mount", f.blobDigest.String(), f.sourceRepoName.Name())
+		opts.Mount.ShouldMount = false
```

Comment:
> is there any case where we would fail a stat and the linkedBlobStore would succeed?

The `linkedBlobStore.Create()` will almost certainly succeed. That cannot be said of the mount though. Nevertheless, since we have the determining knowledge whether to mount or not, we should be explicit about our intention.